cogs/meta.py

```diff
@@ -250,27 +253,61 @@ async def support(self, ctx):
 
 		await ctx.send(self.bot.support_link)
 
-	@commands.command(aliases=['source'])
-	async def code(self, ctx, *, command: str = None):
-		'''Show command code or get GitHub repo link.'''
+	@commands.command(name="source", aliases=["src", "code"])
+	async def source(self, ctx: commands.Context, source_item: str = None):
+		"""Get a github link to the source code of a command."""
 
-		if command is None:
-			await ctx.send('https://github.com/Run1e/AceBot')
+		if source_item is None:
+			await ctx.send(GITHUB_LINK)
 			return
 
-		command = self.bot.get_command(command)
+		# send a typing event while we get from github
+		# normally this would be an async with,
+		# but no reason to do so since it shouldn't take very long to do this code,
+		# therefore no reason to make it with.
+		# if its been long enough that it has to fire again we've errorer or send a response
+		await ctx.trigger_typing()
+
+		cmd: commands.Command = self.bot.get_command(source_item)
 
-		if command is None or command.hidden:
-			raise commands.CommandError('Couldn\'t find command.')
+		if cmd is None:
```

Comment:
Is it possible only allowing commands that would be viewable on github? I.e. use .gitignore or a pygit2.Repository object maybe, like is done here https://github.com/Run1e/AceBot/blob/master/cogs/meta.py#L156-L158

Reply:
Done! It was easier than anticipated.

```diff
+ if pygit2.GIT_STATUS_IGNORED & repo.status_file(source_file):
```